ocs_ci/ocs/platform_nodes.py

```diff
@@ -494,37 +511,50 @@ def restart_nodes(self, nodes, timeout=300, wait=True):
                 and 'ready' state.
 
         """
+        num_events_pre_reboot, num_events_post_reboot = ([] for i in range(2))
         instances = self.get_ec2_instances(nodes)
+        node = nodes[0]
         assert instances, (
             f"Failed to get the EC2 instances for "
             f"nodes {[n.name for n in nodes]}"
         )
+
+        reboot_events_cmd = (
+            f"get events -A --field-selector involvedObject.name="
+            f"{node.name},reason=Rebooted -o yaml"
+        )
+        for node in nodes:
+            num_events_pre_reboot.append(
+                len(node.ocp.exec_oc_cmd(reboot_events_cmd)['items'])
+            )
+
         self.aws.restart_ec2_instances(instances=instances)
+        logger.info("Waiting for 30 seconds...")
+        time.sleep(30)
+
         if wait:
             """
             When reboot is initiated on an instance from the AWS, the
             instance stays at "Running" state throughout the reboot operation.
 
-            Once the OCP node detects that the node is not reachable then the
-            node reaches status NotReady.
-            When the reboot operation is completed and the instance is
-            reachable the OCP node reaches status Ready.
+            When the reboot operation is complete and the instance is reachable
+            the OCP node reaches status Ready and a Reboot event is logged.
             """
             nodes_names = [n.name for n in nodes]
-            logger.info(
-                f"Waiting for nodes: {nodes_names} to reach not ready state"
-            )
-            wait_for_nodes_status(
-                node_names=nodes_names, status=constants.NODE_NOT_READY,
-                timeout=timeout
-            )
             logger.info(
                 f"Waiting for nodes: {nodes_names} to reach ready state"
             )
             wait_for_nodes_status(
                 node_names=nodes_names, status=constants.NODE_READY,
                 timeout=timeout
             )
+            for node in nodes:
+                num_events_post_reboot.append(
+                    len(node.ocp.exec_oc_cmd(reboot_events_cmd)['items'])
```

Comment:
Modify node name in reboot_events_cmd in each iteration

Reply:
Done